src/computation.cpp

```diff
@@ -88,42 +92,181 @@ void Computation::runSimulation()
+
+        // top
+        discretization_->f(discretization_->uIEnd() -1,j) = discretization_->u(discretization_->uIEnd() -1,j);
+    }
+    // set g
+    for ( int j = discretization_->vJBegin(); j < discretization_->vJEnd(); j++)
+    {
+        // left
+        discretization_->g(discretization_->vIBegin() +1,j) = discretization_->v(discretization_->vIBegin() +1,j);
+
+        // top
+        discretization_->g(discretization_->vIEnd() -1,j) = discretization_->v(discretization_->vIEnd() -1,j);
+    }
+
 }
     
 void Computation::computePreliminaryVelocities()
 {
-    return;
+    // calculate F
+    for ( int j = discretization_->uJBegin() +1; j < discretization_->uJEnd() -1; j++)
+    { 
+        for ( int i = discretization_->uIBegin() +1; i < discretization_->uIEnd() -1; i++)
+        {
+            double left_side = discretization_->computeD2uDx2(i,j) + discretization_->computeD2uDy2(i,j);
+            double right_side = - discretization_->computeDu2Dx(i,j) - discretization_->computeDuvDy(i,j);
+            double sum = left_side + right_side + settings_.g[0];
+            discretization_->f(i,j) = discretization_->u(i, j) + dt_ * (1 / settings_.re) * sum;
+        }
+    }
+
+    // calculate G
+    for ( int j = discretization_->vJBegin() +1; j < discretization_->vJEnd() -1; j++)
+    { 
+        for ( int i = discretization_->vIBegin() +1; i < discretization_->vIEnd() -1; i++)
+        {
+            double left_side = discretization_->computeD2vDx2(i,j) + discretization_->computeD2vDy2(i,j);
+            double right_side = - discretization_->computeDuvDx(i,j) - discretization_->computeDv2Dy(i,j);
+            double sum = left_side + right_side + settings_.g[1];
+            discretization_->g(i,j) = discretization_->v(i, j) + dt_ * (1 / settings_.re) * sum;
+        }
+    }
 }
     
 void Computation::computeRightHandSide()
 {
-    return;
+    for ( int j = discretization_->pJBegin() +1; j < discretization_->pJEnd() -1; j++)
+    { 
+        for ( int i = discretization_->pIBegin() +1; i < discretization_->pIEnd() -1; i++)
+        {
+            double difference_f = (discretization_->f(i,j) - discretization_->f(i-1,j)) / meshWidth_[0];
+            double difference_g = (discretization_->g(i,j) - discretization_->g(i,j-1)) / meshWidth_[1];
+            discretization_->rhs(i,j) = (1 / dt_) * (difference_f + difference_g);
+        }
+    }
 }
     
 void Computation::computePressure()
 {
-    return;
+    pressureSolver_.solve();
 }
 
 void Computation::computeVelocities()
 {
-    return;
-}
\ No newline at end of file
+    // calculate u
+    for ( int j = discretization_->uJBegin() +1; j < discretization_->uJEnd() -1; j++)
+    { 
+        for ( int i = discretization_->uIBegin() +1; i < discretization_->uIEnd() -1; i++)
+        {
+            double difference = dt_ * (1/meshWidth_[0]) * (discretization_->p(i+1,j) - discretization_->p(i,j));
```

Comment:
warum berechnest du hier etwas was in discretization schon implementiert ist??

https://github.com/NumerischeSimulation/NumSim/blob/a625b25367e8a26e921e9e4ada60ef84f36ee391/src/discretization/1_discretization.cpp#L43